File: pippip-native-ios/API/NewAccountFinal.m
//
//  NewAccountFinal.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 10/25/17.
//  Copyright © 2017 seComm. All rights reserved.
//

#import "NewAccountFinal.h"
#import "pippip_native_ios-Swift.h"

@interface xNewAccountFinal ()
{
    SessionState *sessionState;
}
@end

@implementation xNewAccountFinal

- (instancetype)init {
    self = [super init];
    
    sessionState = [[SessionState alloc] init];
    return self;
}

- (BOOL)processResponse:(NSDictionary *)response errorDelegate:(id<ErrorDelegate>)errorDelegate {
    
    NSString *errorStr = [response objectForKey:@"error"];
    if (errorStr != nil) {
        [errorDelegate responseError:errorStr];
        return NO;
    }
    else {
        NSString *tokenStr = [response objectForKey:@"authToken"];
        sessionState.authToken = [tokenStr longLongValue];
        return YES;
    }
    
}

@end

File: pippip-native-ios/API/NewAccountFinal.swift
//
//  NewAccountFinal.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 5/30/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit
import ObjectMapper

class NewAccountFinal: NSObject, APIResponseProtocol {
    
    var error: String?
    var sessionId: Int32?
    var authToken: Int64?
    var postId: Int = 0

    var alertPresenter = AlertPresenter()
    var sessionState = SessionState.instance
    
    required init?(map: Map) {
        if map.JSON["sessionId"] == nil {
            return nil
        }
        if map.JSON["authToken"] == nil {
            return nil
        }
    }
    
    func mapping(map: Map) {

        sessionId <- map["sessionId"]
        error <- map["error"]
        authToken <- map["authToken"]
        
    }
    
    func processResponse() throws {

        sessionState.authToken = authToken!
        
    }
    

}
